src/caffeine/prif_private_s.f90

```diff
@@ -238,4 +269,21 @@ pure function optional_value(var) result(c_val)
     end if
   end function
 
+  subroutine caf_establish_child_heap
+    if (caf_this_image(current_team%info%gex_team) == 1) then
+      call caf_allocate_remaining( &
+          current_team%info%heap_mspace, &
+          current_team%info%child_heap_info%allocated_memory, &
+          current_team%info%child_heap_info%size)
+      current_team%info%child_heap_info%offset = &
+          as_int(current_team%info%child_heap_info%allocated_memory) - current_team%info%heap_start
+    end if
+    call prif_co_broadcast(current_team%info%child_heap_info%offset, 1)
+    call prif_co_broadcast(current_team%info%child_heap_info%size, 1)
```

Comment:
Collective operations have a high latency cost, so it will be nearly twice as fast to broadcast the entire `child_team_info` in a single broadcast operation, even if you only need two of the fields.

```suggestion
    call prif_co_broadcast(current_team%info%child_heap_info, 1)
```

This optimization is especially relevant since we currently call this subroutine during every change team or coarray allocation when the current team has children.

Reply:
Good point.